### PR TITLE
feat: NativeAOT iOS support

### LINKED ISSUE #2180: Symbolication not working with real iOS device
### Package

Sentry.Maui

### .NET Flavor

.NET

### .NET Version

7.0.102

### OS

Android

### SDK Version

3.28.1

### Self-Hosted Sentry Version

_No response_

### Steps to Reproduce

1. Make a new MAUI application
2. Add `Sentry.Maui`
3. Write code that throws an exception
4. Configure [MSBuild](https://docs.sentry.io/platforms/dotnet/configuration/msbuild/) with org and project
5. Build in Release mode
6. Deploy to a real iOS device (nott the simulator) 
7. Run the app and cause the exception to fire

### Expected Result

Error in Sentry should include file names and line numbers in the stack trace.

### Actual Result

No symbolication occurs.

image

Event:
https://sentry-sdks.sentry.io/issues/3936606932/events/6292f7f7ec3744e6a410e0bd111879ce/?project=5428537

The JSON of the event does not contain any `debug_meta` information, symbolication is not attempted.

### PR DESCRIPTION
**Pull request by vaind**
Adds support for NativeAOT for .net8-ios (vanilla & MAUI)

closes #2180 

#skip-changelog

### CODE REVIEW

**Comment by tipa** [comment]
Will this also make NativeAOT on macOS work (non-MAUI)? Or should it already work?

**Comment by vaind** [comment]
> Will this also make NativeAOT on macOS work (non-MAUI)? Or should it already work?

That's already available in v4.0.0 alpha

**Comment by tipa** [comment]
For both iOS and macOS? I wasn't able to capture crashes in my NativeAOT iOS app using v4.0.0 alpha

**Comment by vaind** [comment]
> For both iOS and macOS? I wasn't able to capture crashes in my NativeAOT iOS app using v4.0.0 alpha

No, you've asked about macOS so I've answered for macOS. That is already available.

**Comment by vaind** [comment]
> I wasn't able to capture crashes

Also, I've just noticed this bit -> We're currently not handling actual native crashes through our native integration, only managed (.NET) exceptions. In any case, if you have a case that's not captured and should be, please raise an issue, ideally with a reproducible example

**Comment by tipa** [comment]
I was just using `SentrySdk.CauseCrash(CrashType.Managed)` and that didn't show up (only when running in Debug / non-NativeAOT). I'll wait until this PR is merged and a new version is published, then I'll try again.

**Review by bitsandfoxes — APPROVED** [review]
This looks great!

File: samples/Sentry.Samples.Maui/Sentry.Samples.Maui.csproj
Comment (vaind) [reply samples/Sentry.Samples.Maui/Sentry.Samples.Maui.csproj:62]: no such thing as x86_64 iphones

**Pull request merged**
